python/ray/serve/tests/test_proxy_state.py

```diff
@@ -19,13 +22,58 @@
 from ray.serve._private.controller import ServeController
 from ray.serve._private.default_impl import create_cluster_node_info_cache
 from ray.serve._private.proxy import ProxyActor
-from ray.serve._private.proxy_state import ProxyState, ProxyStateManager
+from ray.serve._private.proxy_state import (
+    ActorWrapper,
+    ProxyActorWrapper,
+    ProxyState,
+    ProxyStateManager,
+)
 from ray.serve._private.utils import get_head_node_id
 from ray.serve.config import DeploymentMode, HTTPOptions
 
 HEAD_NODE_ID = "node_id-index-head"
 
 
+class FakeProxyActorWrapper(ActorWrapper):
```

Comment:
@edoakes let me know if this works as what you expected. I will leave other tests unchanged for now so I don't accidentally changed the behavior and hopefully keeping the scope limited. Will refactor other tests to use this as well as a follow up https://github.com/ray-project/ray/issues/39894

Reply:
Looks much better! Why do we need this background asyncio loop though? And why aren't these methods implemented?

This fake should replicate the behavior of the proxy but with manual control. For instance `start_new_ready_check` can increment an internal flag `_ready_checks_started` which is verified in the tests and `is_ready` can return a flag which we can set via another method in the tests.